### DISCUSSION TITLE
Firmware updates

### DISCUSSION THREAD

**Discussion by CTJohnK** [krahabb/meross_lan#197] [General]
Sorry to ask what is most likely a foolish question, but I thought I would ask anyway.

Presuming you have your IoT devices on a VLAN which is limited to that VLAN/internet and cannot access your main network (where HA as well as your other day-to-day devices are used), is there a pressing need to upgrade the Meross firmware?

I have over 30 Meross devices on my IoT VLAN and frankly I don't want the hassle of re-discovering some of those devices after a firmware update.

**Comment by krahabb** [comment]
Hello @CTJohnK,
I guess 99% of the time firmware updates are small fixes (edge cases) and maybe security fixes (I suspect Meross did some of these lately).
So the answer is: if you're good with your devices behaviour and your VLAN is 'safe' you could skip them all.
As for the safety of your VLAN that really depends..I can't really tell. You likely know better than me what the issues might be.

**Comment by nagi603** [comment]
> is there a pressing need to upgrade the Meross firmware?

The need is so not pressing they seem to have currently disabled firmware update detection. I've 3 sets of 4 MSS310 plugs with 3 different firmware versions. The ones I bought about a month ago were updated, but the recent ones are stuck with much earlier (random per box) versions and show as "fully updated".
